pkg/lib/catalogsource/catalogsource_update.go

```diff
@@ -95,6 +95,11 @@ func UpdateSpecAndStatusConditions(logger *logrus.Entry, client versioned.Interf
 		logger.WithError(err).Error("UpdateSpecAndStatusConditions - unable to update CatalogSource image reference")
 		return err
 	}
+
+	if _, err := client.OperatorsV1alpha1().CatalogSources(catsrc.GetNamespace()).UpdateStatus(context.TODO(), catsrc, metav1.UpdateOptions{}); err != nil {
+		logger.WithError(err).Error("UpdateSpecAndStatusConditions - unable to update CatalogSource status condition")
+		return err
+	}
```

Comment:
I think we may want to also check whether this update is even required to avoid unnecessarily bumping the resource version (despite this function's description saying that no checks will be performed) - does subsequent Update -> UpdateStatus calls guarantee that we'll be operating with a different resource version in the latter case? Does a Get need to be threaded in between them?